gio/src/extension.rs

```diff
@@ -0,0 +1,200 @@
+    }
+}
+
+impl<'a> ToGlibPtr<'a, *mut ffi::GIOExtensionPoint> for &'a IOExtensionPoint {
+    type Storage = &'a IOExtensionPoint;
+
+    #[inline]
+    fn to_glib_none(&self) -> Stash<'a, *mut ffi::GIOExtensionPoint, &'a IOExtensionPoint> {
+        Stash(self.0.as_ptr() as *mut ffi::GIOExtensionPoint, *self)
+    }
+}
+
+impl IOExtensionPoint {
+    #[doc(alias = "g_io_extension_point_register")]
+    pub fn register(name: &str) -> Self {
+        unsafe { from_glib_none(ffi::g_io_extension_point_register(name.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "g_io_extension_point_lookup")]
+    pub fn lookup(name: &str) -> Option<Self> {
+        unsafe {
+            let ep = ffi::g_io_extension_point_lookup(name.to_glib_none().0);
+            if ep.is_null() {
+                None
+            } else {
+                Some(from_glib_none(ep))
+            }
+        }
+    }
+
+    #[doc(alias = "g_io_extension_point_get_extensions")]
+    pub fn extensions(&self) -> Vec<IOExtension> {
+        let mut res = Vec::new();
+        unsafe {
+            let mut l = ffi::g_io_extension_point_get_extensions(self.0.as_ptr());
+            while !l.is_null() {
+                let e: *mut ffi::GIOExtension = Ptr::from((*l).data);
+                res.push(from_glib_none(e));
+                l = (*l).next;
+            }
+        }
+        res
+    }
+
+    #[doc(alias = "g_io_extension_point_get_extension_by_name")]
+    pub fn extension_by_name(&self, name: &str) -> Option<IOExtension> {
+        unsafe {
+            let e = ffi::g_io_extension_point_get_extension_by_name(
+                self.0.as_ptr(),
+                name.to_glib_none().0,
+            );
+            if e.is_null() {
+                None
+            } else {
+                Some(from_glib_none(e))
+            }
+        }
+    }
+
+    #[doc(alias = "g_io_extension_point_get_required_type")]
+    pub fn required_type(&self) -> Type {
+        unsafe { from_glib(ffi::g_io_extension_point_get_required_type(self.0.as_ptr())) }
+    }
+
+    #[doc(alias = "g_io_extension_point_set_required_type")]
+    pub fn set_required_type(&mut self, type_: Type) {
+        unsafe { ffi::g_io_extension_point_set_required_type(self.0.as_ptr(), type_.into_glib()) }
+    }
+
+    #[doc(alias = "g_io_extension_point_implement")]
+    pub fn implement(
+        extension_point_name: &str,
+        type_: Type,
+        extension_name: &str,
+        priority: i32,
+    ) -> IOExtension {
+        unsafe {
+            let e = ffi::g_io_extension_point_implement(
+                extension_point_name.to_glib_none().0,
+                type_.into_glib(),
+                extension_name.to_glib_none().0,
+                priority,
+            );
+            from_glib_none(e)
+        }
+    }
+}
+
+/// The implementation of an `IOExtensionPoint`.
+#[doc(alias = "GIOExtension")]
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct IOExtension(ptr::NonNull<ffi::GIOExtension>);
```

Comment:
Should be on a separate file for rustdoc-stripper. Otherwise it won't be documented

Reply:
I did not know! I will split... does it need to follow a specific naming rule? E.g. be called `io_extension.rs` instead of just `extension.rs`?